src/App.tsx

```diff
@@ -1,27 +1,50 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { Good } from './types/Good';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const setGoodsFromServer = (getGoodsFromServer: () => Promise<Good[]>) => {
+    getGoodsFromServer()
+      .then((goodsFromServer: Good[]) => setGoods(goodsFromServer))
+      .catch(error => {
+        throw new Error(error);
+      });
```

Comment:
Why do you need to throw an error inside catch block? I mean, it's good to check if needed values were returned from api (to handle cases when data is returned from api with error), but just to throw same error has no sense